x/ibc/spec/01_concepts.md

```diff
@@ -45,3 +45,28 @@ identifier, but differing feature sets. This will result in undefined behavior
 with regards to version selection in `ConnOpenTry`. Each version in a set of 
 versions should have a unique version identifier.
 :::
+
+### Channel Version Negotation
+
+During the channel handshake procedure a version must be agreed upon between
+the two parties. The selection process is largely left to the callers and 
```

Comment:
As mentioned above, I think the selection process should be done by the channel handler, just using the module-specific callbacks to determine whether to accept/reject a version string at each step in the negotiation.

In other words, the selection algorithm should remain the same; just swapping out black-boxed callbacks for determining version validity/preference.

If we want flexibility over the selection algorithm itself, I think we should still have a default implementation available